keras/src/distribution/tensor_parallel/tensor_layout.py

```diff
@@ -0,0 +1,67 @@
+import collections
+
+from keras.src import ops
+
+
+class Split:
+    """Splits a tensor into shards along a specified dimension.
+
+    This is an internal utility used by a higher-level distribution API.
+    It implements sharding by slicing a tensor along one of its axes.
+    It handles cases where the dimension size is not perfectly divisible by the
+    number of workers by distributing the remainder elements one by one to the
+    first few workers.
+    """
+
+    def __init__(self, device_count, dim, sharding_type="auto"):
+        """Initializes the Split action.
+
+        Args:
+            device_count: The total number of workers/shards.
+            dim: The dimension along which to split the tensor. If -1, the
+                last dimension is used.
+            sharding_type: If `dim` is -1, this can be 'row' (dim=0) or
+                'column' (dim=1) to infer the split axis for 2D tensors.
```

Comment:
Oh we use a different format for documentation. All of this should move inside the documentation of the class line 14.